procrastinate/sql/queries.sql

```diff
@@ -149,3 +149,35 @@ SELECT task_name AS name,
   FROM jobs AS j
  GROUP BY name
  ORDER BY name;
+
+-- list_locks --
+-- Get list of locks and number of jobs per task
+WITH jobs AS (
+   SELECT id,
+          queue_name,
+          task_name,
+          lock,
+          args,
+          status,
+          scheduled_at,
+          attempts
+     FROM procrastinate_jobs
+    WHERE (%(queue_name)s IS NULL OR queue_name = %(queue_name)s)
+      AND (%(task_name)s IS NULL OR task_name = %(task_name)s)
+      AND (%(status)s IS NULL OR status = %(status)s)
+      AND (%(lock)s IS NULL OR lock = %(lock)s)
```

Comment:
When https://github.com/peopledoc/procrastinate/pull/402 is merged I guess we don't want to have NULL locks in the list returned by list_locks, do we?

Reply:
I believe it's already the case ? If `%(lock)s IS NULL`, we don't filter on lock.

Reply:
Ah you mean
```suggestion
      AND (%(lock)s IS NULL OR lock = %(lock)s) AND lock IS NOT NULL
```

Reply:
Yes, exactly.